File: c/sorting_algorithms/notable_sorting_algorithms/bogoSort.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <time.h>

// Funcion para verificar si el array esta ordenado
bool isSorted(int arr[], int n) {
    for (int i = 1; i < n; i++) {
        if (arr[i - 1] > arr[i]) {
            return false;
        }
    }
    return true;
}

// Funcion para permutar aleatoriamente el array
void shuffle(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        int r = rand() % n;
        int temp = arr[i];
        arr[i] = arr[r];
        arr[r] = temp;
    }
}

// Bogo Sort
void bogoSort(int arr[], int n) {
    while (!isSorted(arr, n)) {
        shuffle(arr, n);
    }
}

// Imprimir un array
void printArray(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() {
    srand(time(0));

    int arr[] = {3, 2, 5, 1, 4};
    int n = sizeof(arr) / sizeof(arr[0]);

    printf("\nArray original:\n");
    printArray(arr, n);

    bogoSort(arr, n);

    printf("\nArray ordenado:\n");
    printArray(arr, n);

    return 0;
}

File: python/sorting_algorithms/notable_sorting_algorithms/bogoSort.py
import random

# Funcion para verificar si el array esta ordenado
def is_sorted(arr):
    return all(arr[i] <= arr[i + 1] for i in range(len(arr) - 1))

# Bogo Sort
def bogo_sort(arr):
    while not is_sorted(arr):
        random.shuffle(arr)

# Ejemplo de uso
arr = [3, 2, 5, 1, 4]
print("\nArray original:", arr)
bogo_sort(arr)
print("\nArray ordenado:", arr)